Site Settings: Position site icon setting to left of title, tagline

aduth opened the pull request:
This pull request seeks to reposition the site settings icon field to be floated left of the title and tagline fields for larger viewports.

Before|After
---|---
![Before](https://cloud.githubusercontent.com/assets/1779930/20272911/6afdf8f4-aa5d-11e6-8e48-39b2979bc890.png)|![After](https://cloud.githubusercontent.com/assets/1779930/20448065/1b469fae-adb0-11e6-89bc-ece428a37800.png)

__Testing instructions:__

1. Navigate to [site settings](http://calypso.localhost:3000/settings)
2. Select a site
3. Note that the site icon field is floated to the left of title, tagline fields

cc @mtias

rickybanister commented:
Excited to see this happen. It will be really good to make the site profile settings section like an expanded version of the site card. 

Do you need help with the design, or is that pending?

aduth commented:
@rickybanister Sure! @mtias had tentatively offered to help tweak the layout, but I think he overcommits 😉  I'd be happy to receive some pointers on design direction - I really only had "to the left of title and tagline" to go off of 😆

rickybanister commented:
Well, to get you a little bit further easily, we could:

- ditch the 'description' text for the site icon, the value of the icon seems fairly self-explanatory
- can we wrap the icon in the dotted border we are still using in the old 'site card' edit menu
- allow clicking the icon to engage edit mode
- pad the 'change' button 8px off the icon
- remove the vertical rule between the icon and the other fields
- space the title and description fields 24px right of the site icon

Maybe we see if we can get away with something that simple ^ first?

mtias commented:
Pushed a few tweaks, it's currently like this:

![image](https://cloud.githubusercontent.com/assets/548849/20429741/cd7f1bb2-ad90-11e6-97df-f03a9a9644b5.png)

aduth commented:
Thanks @rickybanister and @mtias ! Looking great now. In 54110e5, I made the icon itself be clickable, taking the same action as the "Change" button. Border styles are shown on hover.

![icon](https://cloud.githubusercontent.com/assets/1779930/20435058/8d4ddc76-ad78-11e6-8c44-ee0018598477.gif)

rickybanister commented:
That feels great! Thanks @aduth and @mtias, this one was a long time coming.

aduth commented:
A little more tweaking, per a comment by @mtias to remove the excess whitespace between icon and title fields:

![image](https://cloud.githubusercontent.com/assets/1779930/20436336/e028c398-ad7d-11e6-86fc-19b86385beca.png)

mtias commented:
This looks great, @aduth. One more thing, though: can we try moving the `i` next to "Site Icon" and make the "Change" button match the width of the icon?

aduth commented:
Tinkered, rebased, squashed, browser tested, engaged in combat with IE11 Flexbox, triumphed... will plan to merge first thing Monday morning.

Final design:

![Site Icon](https://cloud.githubusercontent.com/assets/1779930/20448065/1b469fae-adb0-11e6-89bc-ece428a37800.png)

mtias commented:
Looks great, thanks!
